cpp/dolfinx/fem/Function.h

```diff
@@ -335,7 +336,19 @@ class Function
         xp(0, j) = x(p, j);
 
       // Compute reference coordinates X, and J, detJ and K
-      cmap.compute_reference_geometry(X, J, detJ, K, xp, coordinate_dofs);
+      xt::xtensor<double, 2> Xview = xt::adapt(X.data(), X.shape);
```

Comment:
Declare the xtensor arrays outside of the loop to avoid memory allocations.

Also, these aren't views.

Alternative is to use `auto` and template `compute_reference_geometry`, etc.